### ISSUE TITLE
GUI has random crash

### ISSUE THREAD

**Issue by mdhorn** [clearlinux/clr-installer#381]
**Describe the bug**
Seeing 1 in 10 random crash of the graphical installer due to a segmentation violation. 
Need investigation.

**To Reproduce**
1. Download Live Desktop ISO and flash USB
2. Boot ISO image
3. Completed required options and start installer

**Expected behavior**
Installation completes without crashing.

**Environment (please complete the following information):**
 - Clear Linux OS Version: 29350
 - Installer Version 2.0.3
Not clear if this has been present since 2.0.0 release or not.

**Additional context**
The SEGV appears to be tied into the fact the GTK3 calls are really wrappers around the c-lang GTK bindings.

**Comment by mdhorn** [comment]
See example 
https://gist.github.com/mdhorn/c40c162f28c390d3c3105b4778bfdbb7

**Comment by castulo** [comment]
the gui installer just crashed when I tried running it with this error:
```
Gtk:ERROR:gtkcssnode.c:319:lookup_in_global_parent_cache: assertion failed: (node->cache == NULL)
Aborted
```

**Comment by mdhorn** [comment]
From #435 
@ckrzen commented 6 days ago
Installing to a Lenovo ThinkPad X240 with the clear-29800-live-desktop.iso on USB 3.0 drive:

Here are the errors:

`fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x0]

runtime stack:
runtime.throw(0xb7c9a4, 0x2a)
/usr/lib/golang/src/runtime/panic.go:617 +0x72
runtime.sigpanic()
/usr/lib/golang/src/runtime/signal_unix.go:374 +0x4a9

goroutine 37 [syscall, 8 minutes]:
runtime.cgocall(0x9e9270, 0xc0000e3ad8, 0xc0002bb9e0)
/usr/lib/golang/src/runtime/cgocall.go:128 +0x5b fp=0xc0000e3aa8 sp=0xc0000e3a70 pc=0x46defb
github.com/clearlinux/clr-installer/vendor/github.com/gotk3/gotk3/gtk._Cfunc_gtk_main()
_cgo_gotypes.go:13652 +0x41 fp=0xc0000e3ad8 sp=0xc0000e3aa8 pc=0x7cd161
github.com/clearlinux/clr-installer/vendor/github.com/gotk3/gotk3/gtk.Main(...)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/vendor/github.com/gotk3/gotk3/gtk/gtk.go:943
github.com/clearlinux/clr-installer/gui.(*Gui).Run(0xc000146340, 0xc0000d81e0, 0xc000186020, 0x16, 0x100, 0xc000090de0, 0x17, 0x0, 0x0, 0x0, ...)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/gui/gui.go:97 +0x2b5 fp=0xc0000e3c78 sp=0xc0000e3ad8 pc=0x981025
main.main.func5(0xc0000c5900, 0xc0000d81e0, 0xc000186020, 0x16, 0xc000284830, 0xc000140610, 0xc000172070)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/clr-installer/main.go:317 +0x18d fp=0xc0000e3fa8 sp=0xc0000e3c78 pc=0x9e139d
runtime.goexit()
/usr/lib/golang/src/runtime/asm_amd64.s:1337 +0x1 fp=0xc0000e3fb0 sp=0xc0000e3fa8 pc=0x4c3381
created by main.main
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/clr-installer/main.go:311 +0xdb0

goroutine 1 [chan receive, 8 minutes]:
main.main()
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/clr-installer/main.go:352 +0xe09

goroutine 19 [syscall, 8 minutes]:
os/signal.signal_recv(0x0)
/usr/lib/golang/src/runtime/sigqueue.go:139 +0x9c
os/signal.loop()
/usr/lib/golang/src/os/signal/signal_unix.go:23 +0x22
created by os/signal.init.0
/usr/lib/golang/src/os/signal/signal_unix.go:29 +0x41

goroutine 38 [chan receive, 8 minutes]:
main.main.func6(0xc0001210e0, 0xc0000d81e0, 0xc000172070)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/clr-installer/main.go:344 +0x52
created by main.main
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/clr-installer/main.go:343 +0xdef

goroutine 44 [runnable]:
bufio.ScanLines(0xc0003c326c, 0x7c, 0xd94, 0xd00, 0x7c, 0x0, 0x0, 0x0, 0x0, 0x0)
/usr/lib/golang/src/bufio/scan.go:345 +0x19b
bufio.(*Scanner).Scan(0xc0001bef18, 0xc00008cc80)
/usr/lib/golang/src/bufio/scan.go:145 +0x580
github.com/clearlinux/clr-installer/cmd.RunAndProcessOutput(0xc17b40, 0xc000094180, 0xc0002c4120, 0xb, 0x12, 0xc0002c4120, 0x9)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/cmd/cmd.go:160 +0x29f
github.com/clearlinux/clr-installer/swupd.(*SoftwareUpdater).VerifyWithBundles(0xc0000a4070, 0xb6a016, 0x6, 0x0, 0x0, 0xc0000b60e0, 0xd, 0xe, 0x0, 0x0)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/swupd/swupd.go:309 +0x56c
github.com/clearlinux/clr-installer/controller.contentInstall(0xc000186020, 0x16, 0xb6a016, 0x6, 0xc0000d81e0, 0x100, 0xc000090de0, 0x17, 0x0, 0x0, ...)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/controller/controller.go:486 +0x2ae
github.com/clearlinux/clr-installer/controller.Install(0xc000186020, 0x16, 0xc0000d81e0, 0x100, 0xc000090de0, 0x17, 0x0, 0x0, 0x0, 0x0, ...)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/controller/controller.go:357 +0x1cc7
github.com/clearlinux/clr-installer/gui/pages.(*InstallPage).ResetChanges.func1(0xc0001d4dc0)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/gui/pages/install.go:175 +0x14a
created by github.com/clearlinux/clr-installer/gui/pages.(*InstallPage).ResetChanges
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/gui/pages/install.go:165 +0xf5
`

**Comment by mdhorn** [comment]
From #435 

@dch commented 11 hours ago
me also, I ran clr-installer successfully via terminal with sudo. This is a really serious issue as the installer leaves the disk partly partitioned and largely empty of data depending on it actually crashes.
--

**Comment by mdhorn** [comment]
We believe we have a fix for the these SEVG crashes and it should be in a release soon.

**Comment by mdhorn** [comment]
From #437 

@shenjoon commented 15 hours ago

The installer crashed during "Downloading Required Manifest"

Running pre-install hooks
Testing connectivity
Writing partition table to: sda
Updating partition table for: sda
Adjusting filesystem configurations
Writing vfat file system to sda1 '/boot'
Writing swap file system to sda2
Writing ext4 file system to sda3 '/'
Writing mount files
Resolving OS versions
Cleaning up download directory
Downloading required manifests
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x0]

```
runtime stack:
runtime.throw(0xb7c9a4, 0x2a)
/usr/lib/golang/src/runtime/panic.go:617 +0x72
runtime.sigpanic()
/usr/lib/golang/src/runtime/signal_unix.go:374 +0x4a9

goroutine 12 [syscall]:
runtime.cgocall(0x9e9270, 0xc00019fad8, 0xc000283b00)
/usr/lib/golang/src/runtime/cgocall.go:128 +0x5b fp=0xc00019faa8 sp=0xc00019fa70 pc=0x46defb
github.com/clearlinux/clr-installer/vendor/github.com/gotk3/gotk3/gtk._Cfunc_gtk_main()
_cgo_gotypes.go:13652 +0x41 fp=0xc00019fad8 sp=0xc00019faa8 pc=0x7cd161
github.com/clearlinux/clr-installer/vendor/github.com/gotk3/gotk3/gtk.Main(...)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/vendor/github.com/gotk3/gotk3/gtk/gtk.go:943
github.com/clearlinux/clr-installer/gui.(*Gui).Run(0xc0001343e0, 0xc00000c3c0, 0xc000015000, 0x16, 0x100, 0xc000014ee0, 0x17, 0x0, 0x0, 0x0, ...)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/gui/gui.go:97 +0x2b5 fp=0xc00019fc78 sp=0xc00019fad8 pc=0x981025
main.main.func5(0xc0000b9900, 0xc00000c3c0, 0xc000015000, 0x16, 0xc0002629e8, 0xc000130620, 0xc00005c0e0)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/clr-installer/main.go:317 +0x18d fp=0xc00019ffa8 sp=0xc00019fc78 pc=0x9e139d
runtime.goexit()
/usr/lib/golang/src/runtime/asm_amd64.s:1337 +0x1 fp=0xc00019ffb0 sp=0xc00019ffa8 pc=0x4c3381
created by main.main
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/clr-installer/main.go:311 +0xdb0

goroutine 1 [chan receive]:
main.main()
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/clr-installer/main.go:352 +0xe09

goroutine 5 [syscall]:
os/signal.signal_recv(0x0)
/usr/lib/golang/src/runtime/sigqueue.go:139 +0x9c
os/signal.loop()
/usr/lib/golang/src/os/signal/signal_unix.go:23 +0x22
created by os/signal.init.0
/usr/lib/golang/src/os/signal/signal_unix.go:29 +0x41

goroutine 34 [syscall]:
syscall.Syscall6(0xf7, 0x1, 0x539, 0xc0001ea9e8, 0x1000004, 0x0, 0x0, 0xc0001ea9f8, 0x52d073, 0xc000269f80)
/usr/lib/golang/src/syscall/asm_linux_amd64.s:44 +0x5
os.(*Process).blockUntilWaitable(0xc0000171a0, 0xa, 0x0, 0x10)
/usr/lib/golang/src/os/wait_waitid.go:31 +0x98
os.(*Process).wait(0xc0000171a0, 0xb9b870, 0xb9b878, 0xb9b868)
/usr/lib/golang/src/os/exec_unix.go:22 +0x39
os.(*Process).Wait(...)
/usr/lib/golang/src/os/exec.go:125
os/exec.(*Cmd).Wait(0xc000098840, 0x0, 0x0)
/usr/lib/golang/src/os/exec/exec.go:474 +0x61
os/exec.(*Cmd).Run(0xc000098840, 0x0, 0xc0001eac60)
/usr/lib/golang/src/os/exec/exec.go:318 +0x5c
github.com/clearlinux/clr-installer/cmd.run(0x0, 0x0, 0x0, 0x0, 0xc0001ead40, 0xa, 0xa, 0x4d86da, 0x155e3b0)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/cmd/cmd.go:113 +0x460
github.com/clearlinux/clr-installer/cmd.Run(...)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/cmd/cmd.go:124
github.com/clearlinux/clr-installer/network.FetchRemoteConfigFile(0xc00005c1c0, 0x61, 0x0, 0x0, 0x0, 0x0)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/network/network.go:764 +0x122
github.com/clearlinux/clr-installer/network.DownloadInstallerMessage(0xb6e133, 0x10, 0xb728af, 0x18, 0x0, 0x0)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/network/network.go:780 +0x13b
github.com/clearlinux/clr-installer/gui/pages.(*InstallPage).ResetChanges.func1.1()
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/gui/pages/install.go:170 +0x4b
created by github.com/clearlinux/clr-installer/gui/pages.(*InstallPage).ResetChanges.func1
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/gui/pages/install.go:169 +0x75

goroutine 13 [chan receive]:
main.main.func6(0xc0001112c0, 0xc00000c3c0, 0xc00005c0e0)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/clr-installer/main.go:344 +0x52
created by main.main
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/clr-installer/main.go:343 +0xdef

goroutine 26 [IO wait]:
internal/poll.runtime_pollWait(0x7f43f4640f08, 0x72, 0xffffffffffffffff)
/usr/lib/golang/src/runtime/netpoll.go:182 +0x56
internal/poll.(*pollDesc).wait(0xc000110078, 0x72, 0xd01, 0xdae, 0xffffffffffffffff)
/usr/lib/golang/src/internal/poll/fd_poll_runtime.go:87 +0x9b
internal/poll.(*pollDesc).waitRead(...)
/usr/lib/golang/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc000110060, 0xc0003c3252, 0xdae, 0xdae, 0x0, 0x0, 0x0)
/usr/lib/golang/src/internal/poll/fd_unix.go:169 +0x19b
os.(*File).read(...)
/usr/lib/golang/src/os/file_unix.go:263
os.(*File).Read(0xc000370060, 0xc0003c3252, 0xdae, 0xdae, 0x0, 0x0, 0x0)
/usr/lib/golang/src/os/file.go:108 +0x70
bufio.(*Scanner).Scan(0xc0000d2f18, 0xc00001c680)
/usr/lib/golang/src/bufio/scan.go:213 +0xa4
github.com/clearlinux/clr-installer/cmd.RunAndProcessOutput(0xc17b40, 0xc000110000, 0xc000116120, 0xb, 0x12, 0xc000116120, 0x9)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/cmd/cmd.go:160 +0x29f
github.com/clearlinux/clr-installer/swupd.(*SoftwareUpdater).VerifyWithBundles(0xc0000ca000, 0xb6a016, 0x6, 0x0, 0x0, 0xc0000aa000, 0xb, 0xe, 0x0, 0x0)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/swupd/swupd.go:309 +0x56c
github.com/clearlinux/clr-installer/controller.contentInstall(0xc000015000, 0x16, 0xb6a016, 0x6, 0xc00000c3c0, 0x100, 0xc000014ee0, 0x17, 0x0, 0x0, ...)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/controller/controller.go:486 +0x2ae
github.com/clearlinux/clr-installer/controller.Install(0xc000015000, 0x16, 0xc00000c3c0, 0x100, 0xc000014ee0, 0x17, 0x0, 0x0, 0x0, 0x0, ...)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/controller/controller.go:357 +0x1cc7
github.com/clearlinux/clr-installer/gui/pages.(*InstallPage).ResetChanges.func1(0xc000196af0)
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/gui/pages/install.go:175 +0x14a
created by github.com/clearlinux/clr-installer/gui/pages.(*InstallPage).ResetChanges
/builddir/build/BUILD/clr-installer-2.0.6/.gopath/src/github.com/clearlinux/clr-installer/gui/pages/install.go:165 +0xf5
```

Environment (please complete the following information):

Clear Linux OS Version: [swupd info] 29820
Installer Version [clr-installer -v] 2.0.6
Platform: [hardware infomation] Kabylake NUC

**Comment by mdhorn** [comment]
Released in 2.0.7

**Comment by mdhorn** [comment]
Fix now available in Clear Linux* OS 30000
